### ISSUE TITLE
RoboScape as an Auxiliary Service

### ISSUE THREAD

**Issue by gsteinLTU** [NetsBlox/NetsBlox#2840]
It would be nice if editor and dev (and possibly other instances) could share a single RoboScape port. This would probably require RoboScape to run as an Auxiliary Service, which right now can't send messages (required for RoboScape) and aren't automatically available to all users (pretty much required for RoboScape). So this likely requires some new authentication mechanism, and may require benchmarking to confirm it doesn't add too much overhead on things like robot messages.

**Comment by brollb** [comment]
Agreed. Just to elaborate on some of the design questions mentioned above:
- Could we add support for sending messages from auxiliary services?
    - This will require adding a message sending API to the NetsBlox server.
    - On the security front, here are the main questions that I see:
        - "Should services be able to send messages outside the user's group?"
        - "Could this API be exploited by a malicious service (such as DoS attack)?"
    - For this specific use-case, we actually may be able to side step both of these issues by running the RoboScape service and the server's API behind nginx (where only RoboScape is publicly accessible). **Correction**- This is actually the reasoning behind the current setup with the netsblox services as they no longer run as part of the main server. RoboScape could reuse this same API since it can be hosted on netsblox.org.
- Public Auxiliary Service support.
    - After thinking about this a little more, this might be something that could be added via something like deployment specific auxiliary services.

If RoboScape is being accessed by users of multiple servers, this will require RoboScape to detect which server's API to use for each client request which would need to be updated...
